tools/src/main/scala/org/apache/spark/tools/CheckConnectJvmClientCompatibility.scala

```diff
@@ -182,37 +201,67 @@ class CompatibilitySuite extends ConnectFunSuite {
 
       // RuntimeConfig
       ProblemFilters.exclude[Problem]("org.apache.spark.sql.RuntimeConfig.this"))
-    val problems = allProblems
+    allProblems
       .filter { p =>
         includedRules.exists(rule => rule(p))
       }
       .filter { p =>
         excludeRules.forall(rule => rule(p))
       }
+  }
 
-    if (problems.nonEmpty) {
-      fail(
-        s"\nComparing client jar: $clientJar\nand sql jar: $sqlJar\n" +
-          problems.map(p => p.description("client")).mkString("\n"))
+  private def checkDatasetApiCompatibility(clientJar: File, sqlJar: File): Array[String] = {
```

Comment:
The mima check failed now due to there are 4 incompatible Apis between Dataset in sql module and  Dataset  in connect-client-jvm module with current comparison way, this is the right result.

<img width="1035" alt="image" src="https://user-images.githubusercontent.com/1475305/221600356-f4a555ef-449d-47b1-8ec8-47369009c810.png">